src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php

```diff
@@ -192,43 +190,41 @@ protected function getNumberFormatter(): \NumberFormatter
      */
     private function round(int|float $number): int|float
     {
-        if (null !== $this->scale && null !== $this->roundingMode) {
-            // shift number to maintain the correct scale during rounding
-            $roundingCoef = 10 ** $this->scale;
-
-            if (self::FRACTIONAL == $this->type) {
-                $roundingCoef *= 100;
-            }
+        // shift number to maintain the correct scale during rounding
+        $roundingCoef = 10 ** $this->scale;
 
-            // string representation to avoid rounding errors, similar to bcmul()
-            $number = (string) ($number * $roundingCoef);
-
-            switch ($this->roundingMode) {
-                case \NumberFormatter::ROUND_CEILING:
-                    $number = ceil($number);
-                    break;
-                case \NumberFormatter::ROUND_FLOOR:
-                    $number = floor($number);
-                    break;
-                case \NumberFormatter::ROUND_UP:
-                    $number = $number > 0 ? ceil($number) : floor($number);
-                    break;
-                case \NumberFormatter::ROUND_DOWN:
-                    $number = $number > 0 ? floor($number) : ceil($number);
-                    break;
-                case \NumberFormatter::ROUND_HALFEVEN:
-                    $number = round($number, 0, \PHP_ROUND_HALF_EVEN);
-                    break;
-                case \NumberFormatter::ROUND_HALFUP:
-                    $number = round($number, 0, \PHP_ROUND_HALF_UP);
-                    break;
-                case \NumberFormatter::ROUND_HALFDOWN:
-                    $number = round($number, 0, \PHP_ROUND_HALF_DOWN);
-                    break;
-            }
-
-            $number = 1 === $roundingCoef ? (int) $number : $number / $roundingCoef;
-        }
+        if (self::FRACTIONAL == $this->type) {
+            $roundingCoef *= 100;
+        }
+
+        // string representation to avoid rounding errors, similar to bcmul()
+        $number = (string) ($number * $roundingCoef);
+
+        switch ($this->roundingMode) {
```

Comment:
let's use a match statement for this one

Reply:
I did same on `NumberToLocalizedStringTransformer`